pkg/sac/scope_checker_or.go

```diff
@@ -0,0 +1,208 @@
+			anyAllowedErrs = multierror.Append(anyAllowedErrs, err)
+		}
+	}
+
+	return false, anyAllowedErrs
+}
+
+func (s orScopeChecker) TryAllAllowed(subScopeKeyss [][]ScopeKey) TryAllowedResult {
+	result := Deny
+	for _, checker := range s.scopeCheckers {
+		if res := checker.TryAllAllowed(subScopeKeyss); res == Allow {
+			return res
+		} else if res == Unknown {
+			result = Unknown
+		}
+	}
+	return result
+}
+
+func (s orScopeChecker) AllAllowed(ctx context.Context, subScopeKeyss [][]ScopeKey) (bool, error) {
+	var anyAllowedErrs *multierror.Error
+	for _, checker := range s.scopeCheckers {
+		allowed, err := checker.AllAllowed(ctx, subScopeKeyss)
+		// Short-circuit on the first allowed check result.
+		if allowed {
+			return allowed, nil
+		}
+		if err != nil {
+			anyAllowedErrs = multierror.Append(anyAllowedErrs, err)
+		}
+	}
+
+	return false, anyAllowedErrs
+}
+
+func (s orScopeChecker) ForClusterScopedObject(obj ClusterScopedObject) ScopeChecker {
+	for i := range s.scopeCheckers {
+		s.scopeCheckers[i] = s.scopeCheckers[i].SubScopeChecker(ClusterScopeKey(obj.GetClusterId()))
+	}
+	return s
+}
+
+func (s orScopeChecker) ForNamespaceScopedObject(obj NamespaceScopedObject) ScopeChecker {
+	for i := range s.scopeCheckers {
+		s.scopeCheckers[i] = s.scopeCheckers[i].SubScopeChecker(
+			ClusterScopeKey(obj.GetClusterId()), NamespaceScopeKey(obj.GetNamespace()))
+	}
+	return s
+}
+
+func (s orScopeChecker) AccessMode(am storage.Access) ScopeChecker {
+	for i := range s.scopeCheckers {
+		s.scopeCheckers[i] = s.scopeCheckers[i].SubScopeChecker(AccessModeScopeKey(am))
+	}
+	return s
+}
+
+func (s orScopeChecker) Resource(resource permissions.ResourceHandle) ScopeChecker {
+	for i := range s.scopeCheckers {
+		s.scopeCheckers[i] = s.scopeCheckers[i].SubScopeChecker(ResourceScopeKey(resource.GetResource()))
+	}
+	return s
+}
+
+func (s orScopeChecker) ClusterID(clusterID string) ScopeChecker {
+	for i := range s.scopeCheckers {
+		s.scopeCheckers[i] = s.scopeCheckers[i].SubScopeChecker(ClusterScopeKey(clusterID))
+	}
+	return s
+}
+
+func (s orScopeChecker) Namespace(namespace string) ScopeChecker {
+	for i := range s.scopeCheckers {
+		s.scopeCheckers[i] = s.scopeCheckers[i].SubScopeChecker(NamespaceScopeKey(namespace))
+	}
+	return s
+}
+
+func (s orScopeChecker) Check(ctx context.Context, pred ScopePredicate) (bool, error) {
+	var checkErrs *multierror.Error
+	for _, checker := range s.scopeCheckers {
+		allowed, err := checker.Check(ctx, pred)
+		if allowed {
+			return allowed, nil
+		}
+		if err != nil {
+			checkErrs = multierror.Append(checkErrs, err)
+		}
+	}
+
+	return false, checkErrs
+}
+
+func (s orScopeChecker) EffectiveAccessScope(
+	resource permissions.ResourceWithAccess) (*effectiveaccessscope.ScopeTree, error) {
+	var effectiveAccessScopeErrs *multierror.Error
+	var root = effectiveaccessscope.DenyAllEffectiveAccessScope()
+	for _, checker := range s.scopeCheckers {
+		eas, err := checker.EffectiveAccessScope(resource)
```

Comment:
This should only return [`not supported: use built-in authorizer`](https://github.com/stackrox/stackrox/blob/d653406651df4331a714839ec2c0a23a93425c64/pkg/sac/scope_checker_core_impl.go#L95), you think we can just skip the merging part here @rhybrillou ?